scripts/changelog.sh

```diff
@@ -9,6 +9,10 @@ LATEST_TAG=$(git describe --abbrev=0) # Get the least created tag
 DEFAULT_BRANCH=origin/HEAD
 BASE=$DEFAULT_BRANCH # Change this if you don't want to use the default branch as base
 
+if [ -n "$GITHUB_TOKEN" ]; then
+    echo -n $GITHUB_TOKEN | gh auth login --with-token &> /dev/null
+fi
+
```

Comment:
I don't think the changelog script should handle that setup.

I think it make more sense to have a working env before invoking the script, maybe you can reuse the `setup` command of the GH orb to do the login directly in CircleCI, the script is asserting that we are logged-in with a `gh auth status` and this is the only thing that it should do IMO

https://circleci.com/developer/orbs/orb/circleci/github-cli#usage-setup

Reply:
Hmm yeah now that you said it it may make more sense to fail instead of setuping here

Reply:
You're right, gh/setup does the job as intended